src/ui/preferences/GeneralSettings.qml

```diff
@@ -420,6 +420,14 @@ Rectangle {
                                 enabled:    promptSaveLog.checked && !disableDataPersistence.checked
                                 property Fact _telemetrySaveNotArmed: QGroundControl.settingsManager.appSettings.telemetrySaveNotArmed
                             }
+                            FactCheckBox {
+                                id:         promptSaveCsv
+                                text:       qsTr("Save CSV log of telemetry data")
+                                fact:       _saveCsvTelemetry
+                                visible:    _saveCsvTelemetry && _saveCsvTelemetry.visible
```

Comment:
No need for ```_saveCsvTelemetry &&```. This will never be null.

Reply:
Without it, I get `qrc:/qml/GeneralSettings.qml:427: TypeError: Cannot read property 'visible' of null` when closing QGC. I think that is why the other options have it too.

Reply:
That's correct. I have code identical to that everywhere as those things are indeed NULL during shutdown of the UI.

Reply:
Wow, that's annoying. Hadn't seen those in there. They are still missing in quite a few places. There should be a different way to shutdown to make this not happen. I think we can tear down the Windows first before we close the toolbax and this will go away. Anyway, this is fine then. I'll look at changing shutdown as well.

Reply:
This null crap happens because the QQmlApplicationEngine is not explictly destroyed in the shutdown sequence. So it ends up being deleted last when all the other toolbox stuff is already gone. I fixed it. I'll remove all the null check cruft in that pull.

So @Williangalvani you should pull out the null checks in your changes.

Reply:
Done!